spec/myapp_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+require_relative '../app/models/shop'
+
+describe 'myapp' do
+  include Rack::Test::Methods
+
+  def app
+    Sinatra::Application.new
+  end
+
+  describe 'root path' do
+      let(:shop) {Shop.new({name: 'mandy', description: 'bla'}) }
+      let(:shop) { Shop.create }
```

Comment:
I am having trouble here making a let block equvelent to 
@shop.id = @shop.create

Reply:
@mlan080 I don't completely understand your comment or what you  tried to achieve.... but as far as i understand, as long as you have a a reference to a shop object, you should be good right?

Since your create method returns an id, how about
```
let(:shop) do
  id = Shop.new({name: 'mandy', description: 'bla' }).create
  Shop.find(id)
end
```

Then further in your code

```
expect(response_data).to eq([{ 
  "description"=>"bla", 
  "id"=> shop.id, 
  "name"=>"mandy"
}])
```

Reply:
> with your let block, I would get rid of get '/v1/shops' do
No, that's not true. It only replaces line 12 and 13. line 14 stays

> I also keep getting the error when I run rspec:
> Rack::Test::Error:
> No response yet. Request a page first.
Ofcourse, that's because you removed the `get '/v1/shops'` line. 

So keep 14. Replace 12-13 with the let block i suggested